出品機能実装

ymrt43 opened the pull request:
# What
### フリマアプリのメイン機能である出品機能のサーバーサイド実装
- 現状：単一画像のみの出品機能(プレビュー表示なし)
- gem active-hashでprefecture.rbを作成、商品配送元エリアの選択に使用
- state、fee、termカラムの選択肢は参考元サイトそのままenumにて設定
- 画像データ保存ファイルgitignore追加

# Why
他の機能に繋がる機能であるため、優先的に完了させる

ymrt43 committed:
Create Item model

ghost commented:
Sider detected **1** warning on analyzing the commit 66c695b.

- ⚠️ [Querly](https://sider.review/gh/repos/291402631/pulls/16/logs/3833eb63-de12-448b-877a-bb6419d83f5c)

We recommend fixing them as possible by updating the dependencies, configuring the analysis tool, [configuring `sider.yml`](https://sider.review/gh/repos/291402631/pulls/16#config), [turning off unused tools](https://sider.review/gh/repos/291402631/settings), and so on.

If you have problems or questions still, feel free to ask us via chat. 💬

---

You can [turn off this notification](https://sider.review/gh/repos/291402631/settings#notifications) if you don't want to receive it from now on.

ymrt43 committed:
Create items table

ymrt43 committed:
Add associations with Item model

ymrt43 committed:
Add associations with model User about Item

definition about type of items
on 3 way of conditional branch
1_ buyed_items
2_ sell_items
3_ sold_items

ymrt43 committed:
Add association with item in Category model

ymrt43 committed:
Create images table

ymrt43 committed:
Add association with Item in Image model

ymrt43 committed:
Add mount_uploader in image.rb

ymrt43 committed:
Add validation in image.rb

ymrt43 committed:
Add validation in item.rb

ymrt43 committed:
Item nest with Image model

ymrt43 committed:
Add colon forgotten

ymrt43 committed:
Add @item at items#new

ymrt43 committed:
Add items#create definition

ymrt43 committed:
Add definition of item_params in items_controller.rb

ymrt43 committed:
Add routing of items#create

ymrt43 committed:
Add enum to contents of select_boxes in items#new

ymrt43 committed:
Add selects into  select_box in items#new.html.haml

ymrt43 committed:
Add @categories at items#new

ymrt43 committed:
Change form_helper"select" into "collection_select"

ymrt43 committed:
Delete limit of routes for :items  action

ymrt43 committed:
Set before_action :move_to_index

ymrt43 committed:
gem 'active_hash' install

ymrt43 committed:
Create prefecture.rb as ActiveHash

ymrt43 committed:
Change way of show prefectures selection

enum ->> gem'active_hash'

ymrt43 committed:
Add Prefecture instance

ymrt43 committed:
Merge branch 'master' into Function_item_up_for_aucion

ymrt43 committed:
Add ignored files under public/uploads in .gitignore

ymrt43 committed:
delete validation of :name's uniqueness in user.rb

ymrt43 committed:
Add gem 'pry-rails'

ymrt43 committed:
Add optional:true to buyer_id column & delete unnecessary validations of presence

and delete enum area

ymrt43 committed:
Update items_controller.rb to save item #new

ymrt43 committed:
Delete unnecessary spaces in routes.rb

ymrt43 committed:
delete duplicate gem

akyhysa commented:
LGTM!

shingoteshima commented:
レビュー依頼ありがとうございます！
LGTMです！！
お疲れさまでした！
